python/pyiceberg/manifest.py

```diff
@@ -177,6 +177,12 @@ class DataFile(Record):
     sort_order_id: Optional[int]
     spec_id: Optional[int]
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        # The file_format is written as a string, so we need to cast it to the Enum
+        if name == "file_format":
+            value = FileFormat[value]
```

Comment:
Just curious. Is this related to changes in `pyarrow.py` or just an independent fix for parsing `file_format` in DataFile?

Reply:
This was introduced by merging #7033